src/generic/Poly.jl

```diff
@@ -2743,16 +2743,38 @@ end
 #
 ###############################################################################
 
-function rand(rng::AbstractRNG, S::AbstractAlgebra.PolyRing, deg_range::UnitRange{Int}, v...)
+RandomExtensions.maketype(S::PolyRing, dr::UnitRange{Int}, _...) = elem_type(S)
```

Comment:
`maketype` takes the same arguments as `make`, and must return the type of produced random elements. We could probably define a catch-all method in AA, like `maketype(S::AbstractAlgebra.Set, x...) = elem_type(S)`.